packages/dev/core/src/Loading/sceneLoader.ts

```diff
@@ -577,10 +596,13 @@ export class SceneLoader {
                 onError(request?.statusText, exception);
             };
 
-            const fileOrUrl = fileInfo.file || fileInfo.url;
+            if (!plugin.loadFile && fileInfo.rawData) {
+                throw "Plugin does not support loading ArrayBufferView.";
+            }
+
             request = plugin.loadFile
-                ? plugin.loadFile(scene, fileOrUrl, dataCallback, onProgress, useArrayBuffer, errorCallback)
-                : scene._loadFile(fileOrUrl, dataCallback, onProgress, true, useArrayBuffer, errorCallback);
+                ? plugin.loadFile(scene, fileInfo.rawData || fileInfo.file || fileInfo.url, fileInfo.rootUrl, dataCallback, onProgress, useArrayBuffer, errorCallback, name)
+                : scene._loadFile(fileInfo.file || fileInfo.url, dataCallback, onProgress, true, useArrayBuffer, errorCallback);

```

Comment:
A small note regarding my comment about back-compat - we are already doing it internally here.
Unrelated to this PR - the `request` object is actually never used and is not needed.